source/sampleapps/wifievents_consumer_sample.c

```diff
@@ -103,6 +103,220 @@ int g_disable_csi_log = 0;
+            memcpy(cmd_output + len, buffer, buf_len);
+            len += buf_len;
+            cmd_output[len] = '\0';
+        } else {
+            break;
+        }
+    }
+
+    int status = pclose(fp);
+    if (exit_status) {
+        if (WIFEXITED(status))
+            *exit_status = WEXITSTATUS(status);
+        else
+            *exit_status = -1;
+    }
+
+    return 0;
+}
+
+int get_server_password(char *output_key, size_t output_len)
+{
+    if (!output_key || output_len == 0) {
+        errno = EINVAL;
+        return -1;
+    }
+
+#if defined (_XB7_PRODUCT_REQ_)
+    if (execute_system_command_with_status(
+            "/usr/bin/rdkssacli \"{STOR=GET,SRC=kquhqtoczcbx,DST=/dev/stdout}\"",
+            output_key,
+            output_len,
+            NULL) != 0)
+    {
+        return -1;
+    }
+#else
+    if (system("GetConfigFile /tmp/.cfgDynamicSExpki") != 0) {
+        perror("Failed to generate certificate password");
+        return -1;
+    } else {
+        printf("command:\"GetConfigFile /tmp/.cfgDynamicSExpki\" execute success\r\n");
+    }
+
+    if (execute_system_command_with_status(
+            "cat /tmp/.cfgDynamicSExpki",
+            output_key,
+            output_len,
+            NULL) != 0)
+    {
+        return -1;
+    }
+#endif
+
+    output_key[strcspn(output_key, "\r\n")] = '\0';
+    printf("Key value:%s\r\n", output_key);
+
+    return 0;
+}
+
+int upload_file_to_cloud(const char *file_name)
+{
+    static char password[256] = { 0 };
+    char curl_cmd[1024];
+    char curl_output[1024];
+    int curl_exit_code;
+
+#if defined (_XB7_PRODUCT_REQ_)
+    const char *cert_file_name = "/nvram/certs/devicecert_1.pk12";
+#else
+    const char *cert_file_name = "/nvram/certs/devicecert_2.pk12";
+#endif
+
+    if (!file_name) {
+        fprintf(stderr, "Invalid file name\n");
+        return -1;
+    }
+
+    if (strlen(password) == 0) {
+        if (get_server_password(password, sizeof(password)) != 0) {
+            fprintf(stderr, "Failed to get server password\n");
+            return -1;
+        }
+    }
+
+    for (int attempt = 0; attempt < 2; attempt++) {
+        snprintf(curl_cmd, sizeof(curl_cmd),
+            "curl -s "
+            "--cert-type P12 "
+            "--cert %s:%s "
+            "-F \"data=@%s\" "
+            "https://devprimary.vbautobot.comcast.com:6002/post_csi_file",
+            cert_file_name,
+            password,
+            file_name);
+
+        if (execute_system_command_with_status(
+                curl_cmd,
+                curl_output,
+                sizeof(curl_output),
+                &curl_exit_code) != 0)
+        {
+            fprintf(stderr, "Failed to execute curl\n");
+            return -1;
+        }
+
+        printf("Curl Output:\n%s\n", curl_output);
+
+        if (curl_exit_code == 0) {
+            printf("Upload successful\n");
+            return 0;
+        }
+
+        if (curl_exit_code == 58 && attempt == 0) {
```

Comment:
According to for loop condition there is only one retry:
```suggestion
        if (curl_exit_code == 58) {
```